crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -288,7 +291,229 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::CompleteAuthorizeData
     }
 }
 
-async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
+fn is_multiple_capture_request(
+    multiple_capture_status: Option<enums::CaptureStatus>,
+    capture: Option<storage::Capture>,
+    response: Result<types::PaymentsResponseData, types::ErrorResponse>,
+) -> Option<(enums::CaptureStatus, storage::Capture)> {
+    match response {
+        Ok(_) => multiple_capture_status.zip(capture),
+        Err(_) => capture.map(|capture| (enums::CaptureStatus::Pending, capture)),
+    }
+}
+
+async fn capture_payment_response_update_tracker<F: Clone>(
+    db: &dyn StorageInterface,
+    payment_id: &api::PaymentIdType,
+    mut payment_data: PaymentData<F>,
+    router_data: types::RouterData<F, types::PaymentsCaptureData, types::PaymentsResponseData>,
+    storage_scheme: enums::MerchantStorageScheme,
+) -> RouterResult<PaymentData<F>> {
+    if let Some((capture_status, capture)) = is_multiple_capture_request(
+        router_data.multiple_capture_status,
+        payment_data.capture.clone(),
+        router_data.response.clone(),
+    ) {
+        let (payment_attempt_update, capture_update, connector_response_update) = match router_data
+            .response
+            .clone()
+        {
+            Err(err) => (
+                Some(
+                    storage::PaymentAttemptUpdate::MultipleCaptureResponseUpdate {
+                        status: Some(payment_data.payment_attempt.status),
```

Comment:
There is a payment status lifecycle documentation at the end of this page https://api-reference.hyperswitch.io/docs/hyperswitch-api-reference/3lmsk7ocvq21v-welcome-to-hyperswitch-api-reference, what changes will this feature add to the existing flow diagram?

Reply:
The existing flow won't change. In case of multiple_capture Payment Intent will remain requires_capture until merchant captures the entire amount or merchant decides to to no longer capture the remaining amount(yet to be implemented).

Reply:
in order to reach requires_capture status, capture method can either be `manual` or `manual_multiple`

Reply:
> merchant decides to to no longer capture the remaining amount. 

Please create an issue for this then.